### PR TITLE
assignment

### PR DESCRIPTION
**Pull request by shivakrishna9**

### CODE REVIEW

File: shivakrishna/facebook_login.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    android:background="#3b5998"
+    tools:context="com.example.root.facebook.MainActivity">
+
+<ImageView
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:src="@drawable/fblogo"
+    android:layout_marginTop="70dp"
+    android:id="@+id/logo"/>
+    <TextView
```
Comment (mallelapavank) [inline shivakrishna/facebook_login.xml]: Wrong View

File: shivakrishna/facebook_login.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        android:layout_height="wrap_content"
+        android:id="@+id/forgot"
+        android:layout_below="@id/loginBtn"
+        android:layout_centerInParent="true"
+        android:text="Forgot Password ?"
+        android:textStyle="italic"
+        android:textColor="@android:color/white"/>
+
+
+    <EditText
```
Comment (mallelapavank) [inline shivakrishna/facebook_login.xml]: Why did you use EditText here? You must use TextView to display text. I'm sure you know EditText is used when user input is required as in the case of username and password.

File: shivakrishna/facebook_login.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        android:layout_height="30dp"
+        android:id="@+id/userName"
+        android:background="@android:color/white"
+        android:layout_marginTop="25dp"
+        android:layout_below="@id/logo"
+        android:layout_marginLeft="20dp"
+        android:layout_marginRight="20dp"
+
+        "/>
+    <TextView
```
Comment (mallelapavank) [inline shivakrishna/facebook_login.xml]: Wrong View

**Review by mallelapavank — CHANGES_REQUESTED** [review]
You have used TextView and EditText interchangeably. There is no way for the user to enter Username and Password. And there are some extra double quotes (") which will throw a syntax error. Make these changes asap

File: shivakrishna/facebook_login.xml (hunk 1 above)
Comment (shivakrishna9) [reply shivakrishna/facebook_login.xml]: Sorry for the silly mistakes! I am in a hurry to submit the assignment. I was unable to test it on my phone. It says builds are failing. Will submit it again.

Comment (mallelapavank) [reply shivakrishna/facebook_login.xml]: You don't have to test it on phone everytime. That's why we have AVDs i.e., Android Virtual Devices. It acts like your real phone so you can perform almost all tests there. Have you configured an AVD?

Comment (shivakrishna9) [reply shivakrishna/facebook_login.xml]: Configured AVD. But the builds are failing.  So, now I deleted the configuration folder and restarted again.

Comment (mallelapavank) [reply shivakrishna/facebook_login.xml]: If that din't work delete your existing AVD and create another one. If that fails too send me the exact error message that you are getting

File: shivakrishna/facebook_login.xml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_marginTop="25dp"
+        android:layout_below="@id/userName"
+        android:layout_marginLeft="20dp"
+        android:layout_marginRight="20dp"/>
+    <Button
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="25dp"
+        android:id="@+id/loginBtn"
+        android:layout_below="@id/password"
```
Comment (mallelapavank) [inline shivakrishna/facebook_login.xml]: add `android:inputType="textPassword"`

File: shivakrishna/facebook_login.xml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        android:id="@+id/userName"
+        android:background="@android:color/white"
+        android:layout_marginTop="25dp"
+        android:layout_below="@id/logo"
+        android:layout_marginLeft="20dp"
+        android:layout_marginRight="20dp"/>
+    <EditText
+        android:layout_width="match_parent"
+        android:layout_height="30dp"
+        android:id="@+id/password"
```
Comment (mallelapavank) [inline shivakrishna/facebook_login.xml:28]: add these two attributes
```
android:inputType="textPassword"
android:textColor="@android:color/black"
```

File: shivakrishna/facebook_login.xml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:src="@drawable/fblogo"
+        android:layout_marginTop="70dp"
+        android:id="@+id/logo"/>
+    <EditText
+        android:layout_width="match_parent"
+        android:layout_height="30dp"
+        android:id="@+id/userName"
+        android:background="@android:color/white"
```
Comment (mallelapavank) [inline shivakrishna/facebook_login.xml]: set text color here also

**Comment by M-ZubairAhmed** [comment]
there is no help button too.

**Pull request merged**